pkg/experiment/domain/goal.go

```diff
@@ -39,6 +42,33 @@ func NewGoal(
 	}}, nil
 }
 
+func (g *Goal) Update(
+	name *wrapperspb.StringValue,
+	description *wrapperspb.StringValue,
+	archived *wrapperspb.BoolValue,
+	deleted *wrapperspb.BoolValue,
+) (*Goal, error) {
+	updated := &Goal{}
+	if err := copier.Copy(updated, g); err != nil {
+		return nil, err
+	}
+
+	if name != nil {
+		updated.Goal.Name = name.Value
+	}
+	if description != nil {
+		updated.Goal.Description = description.Value
+	}
+	if archived != nil {
+		updated.Goal.Archived = archived.Value
+	}
+	if deleted != nil {
+		updated.Goal.Deleted = deleted.Value
+	}
```

Comment:
If the goal is not in use, we can delete it from the table. There is no need to do a soft delete.